sqeleton/databases/dremio.py

```diff
@@ -0,0 +1,215 @@
+        return f"rpad(TO_CHAR(CAST({value} AS TIMESTAMP), 'YYYY-MM-DD HH24:MI:SS.{precision_format}'), {TIMESTAMP_PRECISION_POS + 6}, '0')"
+
+    def normalize_number(self, value: str, coltype: FractionalType) -> str:
+        if coltype.precision > 0:
+            value = f"format_number({value}, {coltype.precision})"
+        return f"replace({self.to_string(value)}, ',', '')"
+
+    def normalize_boolean(self, value: str, _coltype: Boolean) -> str:
+        return self.to_string(f"cast({value} as int)")
+
+class Dialect(BaseDialect, Mixin_Schema):
+    name = "Dremio"
+    ROUNDS_ON_PREC_LOSS = False  # False if it truncates.
+    ARG_SYMBOL = None   # Not implemented by Dremio
+    TYPE_CLASSES = {
+        # https://docs.dremio.com/current/reference/sql/data-types/
+        # com.dremio.dac.explore.DataTypeUtil
+        "BOOLEAN": Boolean,
+        "INT": Integer,
+        "INTEGER": Integer,
+        "TINYINT": Integer,  # An alias for Integer.
+        "SMALLINT": Integer,  # An alias for Integer.
+        "BIGINT": Integer,
+        "DECIMAL": Decimal,
+        "DEC": Decimal,  # An alias for Decimal.
+        "NUMERIC": Decimal,  # An alias for Decimal.
+        "FLOAT": Float,
+        "DOUBLE": Float,
+        "REAL": Float,
+        "DATE": Date,
+        "TIME": Timestamp,
+        "TIMESTAMP": Timestamp,  # Dremio does not support timezone-aware timestamps.
+        "CHAR": Text,
+        "CHARACTER": Text,
+        "VARCHAR": Text,
+        "CHARACTER VARYING": Text,  # An alias for VARCHAR.
+        "VARBINARY": Text,
+        "BINARY VARYING": Text,  # An alias for VARBINARY.
+    }
+    MIXINS = {Mixin_Schema, Mixin_MD5, Mixin_NormalizeValue, Mixin_RandomSample}
+
+    def list_tables(self, table_schema: str, like: Compilable = None) -> Compilable:
+        """Query to select the list of tables in the schema. (query return type: table[str])
+
+         If 'like' is specified, the value is applied to the table name, using the 'like' operator.
+         """
+        return (
+            self.table_information()
+            .where(
+                this.table_schema == table_schema,
+                this.table_name.like(like) if like is not None else SKIP,
+            )
+            .select(this.table_name)
+        )
+
+    def timestamp_value(self, t: DbTime) -> str:
+        s = t.isoformat(' ', 'milliseconds')
+        return f"cast('{s}' as timestamp)"
+
+    def quote(self, s: str):
+        return f'"{s}"'
+
+    def to_string(self, s: str):
+        return f"cast({s} as varchar)"
+
+    def parse_type(
+        self,
+        table_path: DbPath,
+        col_name: str,
+        type_repr: str,
+        datetime_precision: int = None,
+        numeric_precision: int = None,
+        numeric_scale: int = None,
+    ) -> ColType:
+        from math import isnan
+        if datetime_precision is not None and not isinstance(datetime_precision, int):
+            if isnan(datetime_precision):
```

Comment:
This should probably be a function (e.g. `nan_to_none()`)

Reply:
Done